### ISSUE TITLE
[BITV] 9.4.1.2/6.2 - All autocomplete combobox elements in the "Profile visibility" section have not been properly implemented with the ARIA, making them not truly accessible for screen reader users. (1)

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/server#41902]
> [!IMPORTANT]
> For QA, see https://github.com/nextcloud/server/issues/41902#issuecomment-1838405393
 

**Please think of adjusting every NcSelect in our code based on new changes in upstream library.**

**ToDo: https://github.com/nextcloud/server/issues/41902#issuecomment-1838405393**

A working example can be found at https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/

![image](https://github.com/nextcloud/server/assets/6078378/fbbb8b8a-b50d-4c14-900d-a51ecccf25a7)

https://report.bitvtest.de/default-en/cbedb5ef-4815-4cba-a83d-995599c178c3.html#checkpoint-bfbf07e5c3-v6-n2

same issues: 
 - https://github.com/nextcloud/server/issues/41905
 - https://github.com/nextcloud/server/issues/41909
 - https://github.com/nextcloud/server/issues/41908
 - https://github.com/nextcloud/server/issues/41893
 - https://github.com/nextcloud/server/issues/41836
 - https://github.com/nextcloud/server/issues/41838

**Comment by JuliaKirschenheuter** [comment]
@susnux could you please observe the implementation of this ticket? Thank you!

**Comment by JuliaKirschenheuter** [comment]
autocomplete is missing?

**Comment by JuliaKirschenheuter** [comment]
`<div>` -> `<input>`?

**Comment by JuliaKirschenheuter** [comment]
same issue: https://github.com/nextcloud/server/issues/41905 ?

**Comment by ShGKme** [comment]
Current issues:
- Requires changes in the [fork of `vue-select`](https://github.com/nextcloud-deps/vue-select/)
  - `role="combobox"` should not be on `<div>`, but on `<input>` directly
  - `aria-label="Search for options"` should not be on this `div[role=combobox]`
    - `aria-label` should be on `input` or `aria-labelledby` / `<label>` should be used
    - The text `"Search for options"` should not be hardcoded and should be translatable
  - The clear button should not have fixed untranslatable `aria-label="Clear Selected"`
  - Wrapper `div` should not have `aria-*`  attributes
- Can be fixed on both fork and `nextcloud/vue` sides
  - Deselect button label and title are not translatable (easier to fix in fork)
  - `aria-labelledby` on `input` is not correct

Not sure if it is an issue:
  - Open indicator probably should be a button, or at least have a label

**Comment by JuliaKirschenheuter** [comment]
Thanks a lor @ShGKme for investigation!

**Comment by szaimen** [comment]
dependency update in stable28 seems to be missing

**Comment by JuliaKirschenheuter** [comment]
no "pending dependency update" anymore. Closing issue

**Comment by szaimen** [comment]
Seems to be present in vue 8.5.0 so should all be available in stable28. However not completely sure how to check all points because they are many...
